boefjes/boefjes/plugins/kat_nikto/main.js

```diff
@@ -0,0 +1,54 @@
+import fs from "node:fs";
+import { execSync } from "node:child_process";
+
+/**
+ * @param {{}} boefje_meta The string input to base64 encode
+ * @returns {(string | string[])[][]}
+ */
+export default function (boefje_meta) {
+  // Depending on what OOI triggered this task, the hostname / address will be in a different location
+  const object_type = boefje_meta.arguments.input.object_type;
+  let ooi = "";
+  if (["IPAddressV4", "IPAddressV6"].includes(object_type))
+    ooi = boefje_meta.arguments.input.address;
+  else if (object_type == "Hostname") ooi = boefje_meta.arguments.input.name;
+  else throw new Error("Unexpected boefje_meta");
+
+  // Running nikto and outputting to a file
+  try {
+    execSync(`./nikto/program/nikto.pl -h ${ooi} -o ./output.json`, {
+      stdio: "inherit",
```

Comment:
We might want to add:
UPDATES=no
To make sure Nikto does not report back to home about unrecognized webservers. (or make this an config argument for the boefje)

Also useful to avoid hanging boefjes:
PROMPTS=NO 

Also useful to be configurable from the boefje settings:
PROXYHOST=
PROXYPORT=
PROXYUSER=
PROXYPASS=
USERAGENT=

Reply:
-404code=301,302,307,308 can be used to make nikto not follow redirects to different hosts.

Reply:
These have been added